src/slq.jl

```diff
@@ -0,0 +1,83 @@
+# Stochastic Lanczos Trace Estimator
+
+using LinearAlgebra
+include("lanczos.jl")
+
+export SLQWorkspace, slq
+
+# predefined function
+invfun(x) = 1/x
+
+mutable struct SLQWorkspace{T, TM<:AbstractMatrix{T}, fn<:Function, rn <: Function,
+    R<:Number, I<:Integer}
+    A::TM
+    f::fn
+    randfunc::rn
+    m::I
+    nᵥ::I
+    ctol::R
+    result::T
+end
+
+"""
+    SLQWorkspace(A; f::Function==invfun, ctol = 0.1, m = 50, nv = 30)
+    SLQWorkspace(A, randfunc::Function; f::Function==invfun, ctol = 0.1, m = 50, nv = 30)
+
+Create SLQWorkspace to work with Stochastic Lanczos Quadrature Algorithm.
+# Arguments
+ - `A` : Symmetric Positive Definite Matrix with low condition number
+ - `f` : Function to apply while calculating trace
+         (If none supplied, f(x) = invfun(X) = 1/x)
+ - `randfunc` : Function to generate arbitary random vectors for Lanczos iteration
+                (If none supplied, f(n) = Base.rand(-1:2:1, n))
+ - `ctol` : Convergence Tolerance
+ - `m` : Lanczos iterations
+ - `nv` : SLQ Iterations (if convergence is achieved SLQ stops before nv)
+"""
+function SLQWorkspace(A; f::Function=invfun, ctol=0.1, m = 50, nv = 30)
+    o = one(eltype(A) <: Integer ? Float64 : eltype(A))
+    randfunc(n) = Base.rand(-o:2*o:o, n)
+    return SLQWorkspace(A, f, randfunc, m, nv, ctol, zero(eltype(A)))
```

Comment:
Should allocate the Lanczos workspace here and have it as a field. The whole point of workspaces is to reuse the memory which is allocated only once at the beginning.